### PR TITLE
[SPARK-34153][SQL] Remove unused `getRawTable()` from `HiveExternalCatalog.alterPartitions()`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
Remove unused call of `getRawTable()` from `HiveExternalCatalog.alterPartitions()`.

### Why are the changes needed?
It reduces the number of calls to Hive External catalog.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
By running the modified test suite:
```
$ build/sbt -Phive-2.3 -Phive-thriftserver "test:testOnly *AlterTableRenamePartitionSuite"
```

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/command/AlterTableRenamePartitionSuite.scala [hunk 1]
```diff
@@ -40,5 +40,13 @@ class AlterTableRenamePartitionSuite
         sql(s"ALTER TABLE $t PARTITION (part=1) RENAME TO PARTITION (part=2)")
       }
     }
+
+    withNamespaceAndTable("ns", "tbl") { t =>
+      sql(s"CREATE TABLE $t (id int, PART int) $defaultUsing PARTITIONED BY (PART)")
+      sql(s"INSERT INTO $t PARTITION (PART=0) SELECT 0")
+      checkHiveClientCalls(expected = 26) {
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/command/AlterTableRenamePartitionSuite.scala:47]: W/o this PR, it is 27

**Comment by MaxGekk** [comment]
@HyukjinKwon @cloud-fan Could you review this, please.

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
@MaxGekk do you want to backport this? It's sort of trivial improvement but it had a conflict in tests so I didn't do that.

**Comment by MaxGekk** [comment]
Here are backports to `branch-3.1`/`branch-3.0`: https://github.com/apache/spark/pull/31241

**Comment by MaxGekk** [comment]
And here is the backport to 2.4: https://github.com/apache/spark/pull/31242

**Comment by HyukjinKwon** [comment]
👍

**Pull request closed without merging**